datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -335,20 +336,24 @@ def populate_options(self, options):
 
         return options
 
-    @property
-    def session(self):
-        if self._session is None:
-            self._session = requests.Session()
+    def _create_session(self):
+        session = requests.Session()
 
-            # Enables HostHeaderSSLAdapter
-            # https://toolbelt.readthedocs.io/en/latest/adapters.html#hostheaderssladapter
-            if self.tls_use_host_header:
-                self._session.mount('https://', host_header_ssl.HostHeaderSSLAdapter())
+        # Enables HostHeaderSSLAdapter
+        # https://toolbelt.readthedocs.io/en/latest/adapters.html#hostheaderssladapter
+        if self.tls_use_host_header:
+            session.mount('https://', host_header_ssl.HostHeaderSSLAdapter())
 
-            # Attributes can't be passed to the constructor
-            for option, value in iteritems(self.options):
-                setattr(self._session, option, value)
+        # Attributes can't be passed to the constructor
+        for option, value in iteritems(self.options):
+            setattr(session, option, value)
```

Comment:
Note: these two lines were not called in the default "don't persist" case previously, so walking through if there's any behavior change here…

* Previously, request-level options were built via the call to `.populate_options()`, which takes the kwargs passed via `http.<method>()`, updates it with `RequestsWrapper` options, and returns the result.
* Now, we do the same thing, except that we initially copy `RequestsWrapper` options onto the `Session` instance (via the two lines highlighted here).

The copying onto the `Session` instance _and_ the behavior of `.populate_options()` is a bit redundant, since we're basically applying defaults twice (once ourselves via `.populate_options()`, and once via `Session` defaults, applied by Requests). But it's not a _broken_ thing to do — and in fact, we were already doing this in the `persist_connections: true` case.

So there doesn't seem to be any behavioral change. Can anyone confirm I'm getting this right?

Reply:
Confirmed